docs/xks/developer-guide/gitops.md

```diff
@@ -0,0 +1,71 @@
+1. As before, this new pull request triggers antoher pipeline that runs the `status` command. This time there is a previous environment and the status command reads the Flux commit status for that environment. Since Flux managed to apply the change in `dev` the `status` command reports success.
+1. The pull request check turns green and the pull request is automatically merged by Git provider.
+1. The Flux Kustomization controller detects that there has been an update to the app's tag in the Git repository and applies this update to the Kubernetes resource for the app (typically a Deployment).
+1. In this case, someoune had applied manual changes to the app's database in the `dev` environment during development. These updates are not present in the `qa` environment, and when the pods running the new container image come up, they cannot read state from the database and so fail their health check. Flux consequently sets a commit status on the `main` branch of the GitHub repository, reporting that the update failed.
+
+Emilia's team has configured Flux to notify them when updates fail and so Emilia's chat client informs her that the update did not go through.
+
+### Application to prod is blocked
+
+<img alt="Apply to prod" src={useBaseUrl("img/assets/xks/developer-guide/developer-flow-apply-prod-blocked.jpg")} />
+
+The workflow for applying to `prod` is similar to that of `qa` above, but since Flux reported failure when applying the update to `qa`, the pipeline running the `status` command will fail and the Git provider will block merging of the pull request.
+
+Seeing that the rollout failed, Emilia investigates and realizes that the release is missing a database migration script. She pushes an updated release tagged `cc2b7e0a` and so triggers the pipline running the `new` command. Because the configuration says `prflow: per-app`, the command "resets" the blocked pull request to apply to the updated release to the `dev` environment.
+
+### Second attempt applying to prod
+
+<img alt="Apply to prod" src={useBaseUrl("img/assets/xks/developer-guide/developer-flow-apply-prod-success.jpg")} />
+
+Emilia's updated app with database migration is successfully applied first to the `dev` environment and then to the `qa` environment. The `status` check for the pull request against `prod` turns green and the pull request can be merged. Since the configuration says `auto: false`, the pull request is not automatically merged. Emilia can now verify the update in the `qa` environment and then merge the pull request through the Git provider's user interface.
```

Comment:
... and then **merges** the pull request through the Git provider's user interface.